src/ott/solvers/linear/univariate.py

```diff
@@ -12,134 +12,247 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Callable, Literal, Optional
+from typing import NamedTuple, Optional, Union
 
 import jax
 import jax.numpy as jnp
 
-from ott.geometry import costs
+from ott.geometry import costs, pointcloud
+from ott.problems.linear import linear_problem
 
-__all__ = ["UnivariateSolver"]
+__all__ = ["UnivariateOutput", "UnivariateSolver"]
+
+
+class UnivariateOutput(NamedTuple):  # noqa: D101
+  prob: linear_problem.LinearProblem
+  ot_costs: float
+  paired_indices: jax.Array
+  mass_paired_indices: jax.Array
+
+  @property
+  def transport_matrices(self) -> jax.Array:
+    """Output a ``[d,n,m]`` tensor of all ``[n,m]`` transport matrices."""
+    assert self.paired_indices is not None, "[d,n,m] Transports *not* computed"
```

Comment:
Based on the typing, `paired_indices` cannot be none, why is this check present?

Reply:
great catch, i have made them optional.